sympy/algebras/quaternion.py

```diff
@@ -39,13 +39,19 @@ def __new__(cls, a=0, b=0, c=0, d=0):
         c = sympify(c)
         d = sympify(d)
 
-        obj = Expr.__new__(cls, a, b, c, d)
+        if((not(a.is_real) and not(a.is_real is None))
```

Comment:
A bit too restrictive, I would allow the case `a.is_real == None`.

Reply:
This check allows one to create a Quaternion(x, x, x, x)
if 
x = symbols('x') or if x = symbols('x', real = True)
but not if
x = symbols('x', real = False) (as in Quaternion(3 + 4*I, 0, 0, 0))

I feel it would be cumbersome for the user to set real = True everytime.